core/src/main/scala/org/apache/spark/ui/UIUtils.scala

```diff
@@ -527,4 +528,25 @@ private[spark] object UIUtils extends Logging {
       origHref
     }
   }
+
+  /**
+   * Remove suspicious characters of user input to prevent Cross-Site scripting (XSS) attacks
+   *
+   * For more information about XSS testing:
+   * https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet and
+   * https://www.owasp.org/index.php/Testing_for_Reflected_Cross_site_scripting_(OTG-INPVAL-001)
+   */
+  def stripXSS(requestParameter: String): String = {
+    var requestParameterStripped = requestParameter
```

Comment:
I think the body is simpler as

```
if (requestParameter == null) {
  null
} else {
  StringEscapeUtils.escapeHtml4(
    requestParameter.replaceAll("(\r\n|\n|\r|%0D%0A|%0A|%0D|'|%27)", ""))
}
```

I'm not sure if it will matter for performance but you could pull out the regex as a saved instance variable instead of compiling it each time.